apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -168,56 +169,7 @@ function handlePressThis( calypsoPort ) {
 		}
 
 		calypsoPort.removeEventListener( 'message', onPressThis, false );
-
-		const unsubscribe = subscribe( () => {
-			// Calypso sends the message as soon as the iframe is loaded, so we
-			// need to be sure that the editor is initialized and the core blocks
-			// registered. There is an unstable selector for that, so we use
-			// `isCleanNewPost` otherwise which is triggered when everything is
-			// initialized if the post is new.
-			const editorIsReady = select( 'core/editor' ).__unstableIsEditorReady
-				? select( 'core/editor' ).__unstableIsEditorReady()
-				: select( 'core/editor' ).isCleanNewPost();
-			if ( ! editorIsReady ) {
-				return;
-			}
-
-			unsubscribe();
-
-			const title = get( payload, 'title' );
-			const text = get( payload, 'text' );
-			const url = get( payload, 'url' );
-			const image = get( payload, 'image' );
-			const embed = get( payload, 'embed' );
-			const link = `<a href="${ url }">${ title }</a>`;
-
-			const blocks = [];
-
-			if ( embed ) {
-				blocks.push( createBlock( 'core/embed', { url: embed } ) );
-			}
-
-			if ( image ) {
-				blocks.push(
-					createBlock( 'core/image', {
-						url: image,
-						caption: text ? '' : link,
-					} )
-				);
-			}
-
-			if ( text ) {
-				blocks.push(
-					createBlock( 'core/quote', {
-						value: `<p>${ text }</p>`,
-						citation: link,
-					} )
-				);
-			}
-
-			dispatch( 'core/editor' ).resetEditorBlocks( blocks );
-			dispatch( 'core/editor' ).editPost( { title: title } );
-		} );
+		doPressThis( payload );
```

Comment:
Features are not meant to be shared across bundles, because we already have a `default` bundle which does exactly that (provides features to both the iframed editor and the WP Admin editor).

Since we're already adding the feature to the `default` bundle,  we should be able to remove the logic from the `calypso` bundle.

We would need to pass the press this query params (`url`, `title`, `text`, etc) captured by Calypso to the iframe source though:

https://github.com/Automattic/wp-calypso/blob/56c5b5b3892f0d5e429f8377dea75920f422bf4e/client/gutenberg/editor/calypsoify-iframe.tsx#L822

Could be done directly based on the existing `pressThis` prop:

```diff
diff --git a/client/gutenberg/editor/calypsoify-iframe.tsx b/client/gutenberg/editor/calypsoify-iframe.tsx
index 6c78f6083f..b41407f7e7 100644
--- a/client/gutenberg/editor/calypsoify-iframe.tsx
+++ b/client/gutenberg/editor/calypsoify-iframe.tsx
@@ -811,6 +811,7 @@ const mapStateToProps = (
                stripeConnectSuccess,
                anchorFmData,
                showDraftPostModal,
+               pressThis,
        }: Props
 ) => {
        const siteId = getSelectedSiteId( state );
@@ -836,6 +837,7 @@ const mapStateToProps = (
                ...anchorFmData,
                openSidebar: getQueryArg( window.location.href, 'openSidebar' ),
                showDraftPostModal,
+               ...pressThis,
        } );
 
        // needed for loading the editor in SU sessions
```

That would enable us to simplify the overall implementation by removing some chunks of code like [the `pressThis` function](https://github.com/Automattic/wp-calypso/blob/56c5b5b3892f0d5e429f8377dea75920f422bf4e/client/gutenberg/editor/calypsoify-iframe.tsx#L612-L623) or [the `handlePressThis` function](https://github.com/Automattic/wp-calypso/blob/32cced72bbd5d4a7f4569da7c2360f4550534e08/apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js#L160-L174).

Reply:
Thank you, this worked a treat!

I tested this works, cleaned up the other code, and updated the readme, thanks!